src/app/js/components/column/messageSender/MessageSenderView.js

```diff
@@ -67,18 +68,22 @@ class MessageSender extends Component {
         const state = this.state
         const messageCollection = state.messageCollection.slice()
         messageCollection[state.messageInEditor] = newValue
+        const messageIsJson = this.jsonOrText(newValue)
+        const messageIsJsonCollection = state.messageIsJsonCollection
+        messageIsJsonCollection[this.props.activeTab] = messageIsJson
```

Comment:
You're mutating the state here. To prevent this you should first make a copy of the original 'messageIsJsonCollection' object.